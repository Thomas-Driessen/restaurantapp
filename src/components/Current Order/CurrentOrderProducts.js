import React from 'react';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import RemoveIcon from '@material-ui/icons/Remove';
import Grid from '@material-ui/core/Grid';

const CurrentOrderProducts = (props) => {

    return (
        <div>
            { props.product ? (
                <div>
                    <Card >
                        <CardContent>
                            <Typography gutterBottom variant="inherit" component="h2">
                                {props.product.title} <span style={{ float: "right", color: "green" }}>{props.product.price}â‚¬</span>
                            </Typography>
                            <Typography component="h6">
                                {props.product.ingredients}
                            </Typography>
                        </CardContent>
                        <CardActions>
                            <Grid container alignItems="flex-start" justify="flex-end" direction="row">
                                <IconButton aria-label="remove from order" onClick={() => props.remove(props.product)}>
                                    <RemoveIcon />
                                </IconButton>
                            </Grid>
                        </CardActions>
                    </Card>
                </div>
            ) : null}
        </div>
    )
}
export default CurrentOrderProducts