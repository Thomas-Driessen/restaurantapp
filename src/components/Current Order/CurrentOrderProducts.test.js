import React from 'react';
import { render } from '@testing-library/react';
import CurrentOrderProducts from './CurrentOrderProducts';
import renderer from 'react-test-renderer';

const product = {
  id: 1,
  title: "Coca-Cola",
  price: 2.5,
  description: "",
  ingredients: [
    {ingredient:[
      {IngredientTitle: "Water"}
    ]}
  ]
};

it('CurrentOrderProducts render correctly', () => {
  const tree = renderer
    .create(<CurrentOrderProducts product={product} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render CurrentOrderProducts', () => {
  const { getByText } = render(<CurrentOrderProducts product={product} />);
  const linkElement = getByText(/Coca-cola/i);
  expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
  global.gc && global.gc()
})