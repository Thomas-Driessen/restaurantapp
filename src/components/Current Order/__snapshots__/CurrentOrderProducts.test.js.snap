// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentOrderProducts render correctly 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-gutterBottom"
        >
          Coca-Cola
           
          <span
            style={
              Object {
                "color": "green",
                "float": "right",
              }
            }
          >
            2.5
            â‚¬
          </span>
        </h2>
        <h6
          className="MuiTypography-root MuiTypography-body1"
        >
          
        </h6>
      </div>
      <div
        className="MuiCardActions-root MuiCardActions-spacing"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
        >
          <button
            aria-label="remove from order"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13H5v-2h14v2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
